File: src/main/java/me/onixdev/util/net/ClientInput.java
package me.onixdev.util.net;

import dev.onixac.api.user.IClientInput;

public class ClientInput implements IClientInput {
    private boolean forward;
    private boolean backward;
    private boolean left;
    private boolean right;
    private boolean jump;
    private boolean shift;
    private boolean sprint;
    private final boolean slowing;
    private double forwardMotion;
    private double strafe;
    private final double motion;

    public ClientInput(boolean forward, boolean backward, boolean left, boolean right, boolean jump, boolean shift, boolean sprint,boolean slowing, double forwardMotion, double strafe,double motion) {
        this.forward = forward;
        this.backward = backward;
        this.left = left;
        this.right = right;
        this.jump = jump;
        this.shift = shift;
        this.sprint = sprint;
        this.forwardMotion = forwardMotion;
        this.strafe = strafe;
        this.motion = motion;
        this.slowing = slowing;
    }

    public ClientInput() {
        this(false, false, false, false, false, false, false,false,0,0,0);
    }

    public boolean isForward() {
        return forward;
    }

    public boolean isBackward() {
        return backward;
    }

    public boolean isLeft() {
        return left;
    }

    public boolean isRight() {
        return right;
    }

    public boolean isJump() {
        return jump;
    }

    public boolean isShift() {
        return shift;
    }

    public boolean isSprint() {
        return sprint;
    }
    public boolean hasInput() {
        return forward || backward || left || right;
    }


    public double getForwardMotion() {
        return forwardMotion;
    }

    public double getStrafe() {
        return strafe;
    }

    public boolean isSlowing() {
        return slowing;
    }

    public void reset() {
        forward = backward = left = right = jump = shift = sprint = false;
        forwardMotion = strafe = 0;
    }


    public double getMotion() {
        return motion;
    }

    @Override
    public String toString() {
        return "ClientKeyInput{" +
                "forward=" + forward +
                ", backward=" + backward +
                ", left=" + left +
                ", right=" + right +
                ", jump=" + jump +
                ", shift=" + shift +
                ", sprint=" + sprint +
                '}';
    }
}

File: src/main/java/me/onixdev/util/net/ClientInput.kt
package me.onixdev.util.net

import dev.onixac.api.user.IClientInput

class ClientInput @JvmOverloads constructor(
    private var forward: Boolean = false,
    private var backward: Boolean = false,
    private var left: Boolean = false,
    private var right: Boolean = false,
    private var jump: Boolean = false,
    private var shift: Boolean = false,
    private var sprint: Boolean = false,
    private val slowing: Boolean = false,
    private var forwardMotion: Double = 0.0,
    private var strafe: Double = 0.0,
    private val motion: Double = 0.0
) : IClientInput {
    override fun isForward(): Boolean {
        return forward
    }

    override fun isBackward(): Boolean {
        return backward
    }

    override fun isLeft(): Boolean {
        return left
    }

    override fun isRight(): Boolean {
        return right
    }

    override fun isJump(): Boolean {
        return jump
    }

    override fun isShift(): Boolean {
        return shift
    }

    override fun isSprint(): Boolean {
        return sprint
    }

    override fun hasInput(): Boolean {
        return forward || backward || left || right
    }


    override fun getForwardMotion(): Double {
        return forwardMotion
    }

    override fun getStrafe(): Double {
        return strafe
    }

    override fun isSlowing(): Boolean {
        return slowing
    }

    override fun reset() {
        sprint = false
        shift = sprint
        jump = shift
        right = jump
        left = right
        backward = left
        forward = backward
        strafe = 0.0
        forwardMotion = strafe
    }


    override fun getMotion(): Double {
        return motion
    }

    override fun toString(): String {
        return "ClientKeyInput{" +
                "forward=" + forward +
                ", backward=" + backward +
                ", left=" + left +
                ", right=" + right +
                ", jump=" + jump +
                ", shift=" + shift +
                ", sprint=" + sprint +
                '}'
    }
}